### ISSUE TITLE
Support Vector Machine to classify protein-protein interfaces

### ISSUE THREAD

**Issue by romulogoncalves** [NLeSC/TEAM2018#42]
@NicoRenaud could you provide a bit more info about the paper you will work on? Would some help be welcomed or do you plan to work alone on it?

**Comment by NicoRenaud** [comment]
@romulogoncalves the paper is on a graph-based method using Support Vector Machine to classify protein-protein interfaces. 

The software compute graphs of protein interfaces, kernels between pairs of graphs and use the kernels to train a model with SVM (using libsvm under the hood). The model can then used to predict the quality of new protein interfaces. A simple command line interface greatly facilitate the workflow and can compute graphs and kernels in parallel (using mpi4py).

The computation of the graph kernel can be done with a (rather simple) GPU implementation that I'm still trying to improve (the Kernel Tuner can be used to fine tune there).

I've worked on that project only with people from Utrecht so I would rather work alone on it.

**Comment by romulogoncalves** [comment]
@NicoRenaud  is there project hours for this work? If yes, which one?

**Comment by NicoRenaud** [comment]
The project is DeepRank, there is still plenty of hours and I checked with Elena who's the project coordinator and she's ok with me spending a few hours writing the software paper.

**Comment by romulogoncalves** [comment]
@NicoRenaud were all the goals achieved? Could you summarize in 3 lines what was achieved? Can we consider it done and close this issue?

**Comment by NicoRenaud** [comment]
Output of the Sprint:

1 - The draft of the software paper is in a correct state but not finished. I will beed a few more days to finalize it. I hope to finish it before the Christmas holidays. 

2- The code itself has been improved in terms of quality, features and usability. 

3 - A simple GPU kernel using shared memory for computing the random walk graph kernel has been written leading to a very small performance improvement for very large graphs.

4 - The different solutions for computing the random walk graph kernel have been benchmarked.
